config/config_partial.go

```diff
@@ -457,11 +457,17 @@ func PartialParseConfig(ctx *ParsingContext, file *hclparse.File, includeFromChi
 			}
 		case TerragruntInputs:
 			allControls := ctx.TerragruntOptions.StrictControls
-			skipDependenciesInputs := allControls.FilterByNames(controls.SkipDependenciesInputs)
-			logger := log.ContextWithLogger(ctx, ctx.TerragruntOptions.Logger)
 
+			skipDependenciesInputs := allControls.Find(controls.SkipDependenciesInputs)
+			if skipDependenciesInputs == nil {
+				return nil, errors.New("failed to find control " + controls.SkipDependenciesInputs)
+			}
+
+			skipDependenciesInputs.SuppressWarning()
```

Comment:
`skipDependenciesInputs` is a pointer. If you change it, it will also affect `opts.Strict Controls`.
It would be better not to modify `opts.StrictControls` at all. In another place where you use it, it will be unclear why the value is different from what is specified by default. This may especially surprise other developers who are not aware.

Reply:
Ya, I would prefer that we not mutate the control, but we already do it for `Enable()`, and I think it's better to keep things consistent. We shouldn't _sometimes_ mutate in methods of pointer receivers.

We should either always do that, and have that be the expectation, or never do it.

Reply:
In some ways you are right, but these are slightly different things :) Enable is done to configure the controls before starting work, and this is definitely something the user does explicitly. But in this case, it's like a magic. If you think it's better this way, so be it.